### PR TITLE
Genlex refactoring

### PR DESCRIPTION
**Pull request by d-plaindoux**
Remove Lexer Token module and extend Lexeme module

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,38 +120,38 @@ like `constant`. The `CharParser` module is a is parser dedicated to char stream
 parsing using another parser.
 
 ```ocaml
-module Utils = Transept_utils.Utils
-module CharParser = Transept_extension.Parser.For_char_list
-module Stream = Transept_stream.Via_parser (CharParser)
-module Genlex = Transept_genlex.Genlex.Make (CharParser)
+module Utils = Transept.Utils.Utils
```
Comment (xvw) [inline README.md:123]: I think `Transept.Utils.Utils` seems a little bit strange. I slightly preferer having `Transpet.Utils`

File: README.md [hunk 2]
```diff
@@ -120,38 +120,38 @@ like `constant`. The `CharParser` module is a is parser dedicated to char stream
...
+module CharParser = Transept.Extension.Parser.For_char_list
+module Stream = Transept.Stream.Via_parser (CharParser)
+module Genlex = Transept.Genlex.Genlex.Make (CharParser)
```
Comment (xvw) [inline README.md:126]: Same here for `Transept.Genlex.Genlex`.

**Review by xvw — COMMENTED** [review]
I prefer your proposition (coupling lexeme/token).
I find that the process of building a Stream is still a little bit painful. But I don't have alternative.

File: README.md (hunk 1 above)
Comment (d-plaindoux) [reply README.md:123]: Utils has been renamed as Fun.

File: README.md (hunk 2 above)
Comment (d-plaindoux) [reply README.md:126]: Genlex has been been renamed as Lexer

Comment (xvw) [reply README.md:126]: In which commit? In the text I keep seeing "Genlex.Genlex"

**Review by xvw — APPROVED** [review]
nice!

**Pull request merged**